ga4gh/datamodel/reads.py

```diff
@@ -289,6 +300,23 @@ def getNumUnalignedReads(self):
     def getPrograms(self):
         return self._programs
 
+    def getReadAlignments(self, reference, start, end):
+        """
+        Returns an iterator over all reads in a ReadGroupSet
+        """
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        syntheticReadGroup = HtslibReadGroup(self, 'synthetic')
```

Comment:
Yeah, I think that's probably a good approach. Say, ``AlignmentDataMixin`` which subclasses ``datamodel.PysamDatamodelMixin``? Creating a new readgroup here will complicate things in the long run as we're re-interpreting the header information for each incoming request.